### DISCUSSION TITLE
ESP8266 I2Splay disabled?

### DISCUSSION THREAD

**Discussion by Tinky1233** [arendst/Tasmota#19376] [Support request] [answered]
Why is I2SPlay command disabled for ESP8266 chip. It is perfectly capable of streaming and playing mp3 as demonstrated in https://github.com/schmurtzm/MrDiy-Audio-Notifier. Why not use inexpensive nodemcu v3 board that has 4MB flash, to store and play simple notification sounds, ESP32 has smaller filesystem on 4MB versions due to larger program size than ESP8266. I think ESP8266 is more usefull and cost efective to play simple mp3 sounds.

**Comment by pkkrusty** [comment, marked as answer]
I spent some time trying to get i2splay to work on esp8266, and I think with all the other overhead that Tasmota has, there isn't enough leftover RAM to do it. Even tried with very low bitrates with no success. I think it's probably a bridge too far, since if @gemu2015  couldn't make it work, it's probably not possible.

**Comment by sfromis** [comment]
The fact that some function may be workable on ESP8266 standalone, taking up a major part of RAM on that resource-constrained chip, does not mean that it would fit as just one of the wide array of features Tasmota supports.

**Comment by Tinky1233** [comment]
Ok, can it at least play short wav files from filesystem, does that use less RAM? Adding SD Card can be alternative?

**Reply by pkkrusty** [reply]
Good question. I would think wav would be less resource intensive. Would be a very useful feature, I agree.

**Reply by gemu2015** [reply]
i checked esp8266 with i2s audio features when i implemented i2s support.
at that time i also had wav support in place. however wav files are much larger then mp3.
over all memory constraints were so severe that i disabled esp8266 support.

i however love sounds on my tasmota devices and for some of my esp8266 i use external mp3 hardware, especially 
DY_SV17F, this one has only 4MB Flash instead of SD card (which is enough for dozens of sounds) but you may play sounds by name not by index and it is below 2 Euros.
